lib/wasix/src/runtime/mod.rs

```diff
@@ -76,6 +79,63 @@ where
     fn tty(&self) -> Option<&(dyn TtyBridge + Send + Sync)> {
         None
     }
+
+    /// Load a a Webassembly module, trying to use a pre-compiled version if possible.
+    fn load_module<'a>(&'a self, wasm: &'a [u8]) -> BoxFuture<'a, Result<Module, anyhow::Error>> {
```

Comment:
I'm sure we'll miss one or two places were we run `Module::new()`, but I'm happy with this addition 👍 

It also helps with #3985 because now we've got something that can be hooked into for showing a "Compiling..." spinner.